src/STW.ProcessingApi.Function/Rules/Interfaces/IAsyncRule.cs

```diff
@@ -0,0 +1,10 @@
+namespace STW.ProcessingApi.Function.Rules.Interfaces;
+
+using Models;
+
+public interface IAsyncRule
+{
+    bool ShouldInvoke(SpsCertificate spsCertificate);
+
+    Task InvokeAsync(SpsCertificate certificate, IList<ValidationError> errors);
```

Comment:
You called this one `certificate` and the one above `spsCertificate`. Go with `spsCertificate` for both.